### PR TITLE
HPUPrecision plugin with autocast

### PR DESCRIPTION
**Pull request by jerome-habana**
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

Fixes # (issue).

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/lightning_habana/fabric/plugins/precision.py [hunk 1]
```diff
@@ -12,66 +12,53 @@
...
-from lightning_habana import _HPU_AVAILABLE
-from lightning_habana.utils.imports import _HABANA_FRAMEWORK_AVAILABLE
-
-if _HABANA_FRAMEWORK_AVAILABLE:
-    from habana_frameworks.torch.hpex import hmp
-
-_PRECISION_INPUT_INT = Literal[32]
-_PRECISION_INPUT_STR = Literal["32", "bf16", "32-true", "bf16-mixed"]
-_PRECISION_INPUT = Union[_PRECISION_INPUT_INT, _PRECISION_INPUT_STR]
+_PRECISION_INPUT = Literal["32", "bf16", "32-true", "bf16-mixed"]
```
Comment (Borda) [inline src/lightning_habana/fabric/plugins/precision.py:30]: Shall these be with `?-true` sufix?

Comment (jerome-habana) [reply src/lightning_habana/fabric/plugins/precision.py:30]: might be problematic when other types are supported & untested

Comment (jerome-habana) [reply src/lightning_habana/fabric/plugins/precision.py:30]: @Borda - can we close this ?

File: src/lightning_habana/fabric/plugins/precision.py [hunk 2]
```diff
@@ -12,66 +12,53 @@
...
     def __init__(
         self,
         precision: _PRECISION_INPUT,
-        opt_level: str = "O2",
-        bf16_file_path: Optional[str] = None,
-        fp32_file_path: Optional[str] = None,
-        verbose: bool = False,
+        device: str = "hpu",
     ) -> None:
-        rank_zero_info(
```
Comment (Borda) [inline src/lightning_habana/fabric/plugins/precision.py:60]: why is this already removed?

**Review by Borda — COMMENTED** [review]
I think we shall have to check for the Habana version because this may not work with 1.10, right?

File: src/lightning_habana/fabric/plugins/precision.py (hunk 2 above)
Comment (jerome-habana) [reply src/lightning_habana/fabric/plugins/precision.py:60]: From 1.11.0 synpase AI release, this is supported fully. Asd we have already changed the default docker image, its removed

**Comment by jerome-habana** [comment]
> I think we shall have to check for the Habana version because this may not work with 1.10, right?

Yep. It wont work with 1.10. But the support matrix should take care of it.

**Comment by Borda** [comment]
> Yep. It wont work with 1.10. But the support matrix should take care of it.

right, but some/most users are lazy, so they won't read it and we shall include it in the code oat least as a warning

File: src/lightning_habana/fabric/plugins/precision.py (hunk 2 above)
Comment (jerome-habana) [reply src/lightning_habana/fabric/plugins/precision.py:60]: updated logs

File: src/lightning_habana/pytorch/plugins/precision.py [hunk 3]
```diff
@@ -25,45 +28,41 @@
...
-        bf16_file_path: Optional[str] = None,
-        fp32_file_path: Optional[str] = None,
-        verbose: bool = False,
+        device: str = "hpu",
     ) -> None:
         rank_zero_info(
-            "The 'HMP' support is deprecated and will be removed in lightning-habana release 1.12.0,"
-            " Use 'torch autocast' instead."
+            "The 'HMP' support is deprecated and will be removed in lightning-habana release 1.1.0,"
+            " Use 'torch autocast' instead. or use lightning-habana <= 1.0.1"
```
Comment (Borda) [inline src/lightning_habana/pytorch/plugins/precision.py:50]:
```suggestion
            " Use 'torch autocast' instead. or use lightning-habana < 1.1.0"
```

File: src/lightning_habana/fabric/plugins/precision.py [hunk 4]
```diff
@@ -27,51 +29,42 @@
...
-        bf16_file_path: Optional[str] = None,
-        fp32_file_path: Optional[str] = None,
-        verbose: bool = False,
+        device: str = "hpu",
     ) -> None:
         rank_zero_info(
-            "The 'HMP' support is deprecated and will be removed in lightning-habana release 1.2.0,"
-            " Use 'torch autocast' instead."
+            "The 'HMP' support is deprecated and will be removed in lightning-habana release 1.1.0,"
+            " Use 'torch autocast' instead. or use lightning-habana <= 1.0.1"
```
Comment (Borda) [inline src/lightning_habana/fabric/plugins/precision.py:50]:
```suggestion
            " Use 'torch autocast' instead. or use lightning-habana < 1.1.0"
```

File: README.md [hunk 5]
```diff
@@ -63,12 +63,12 @@ The `devices>1` parameter with HPUs enables the Habana accelerator for distribut
...
 
-| **SynapseAI**         | **1.10.0**                                         |
+| **SynapseAI**         | **1.11.0**                                         |
 | --------------------- | -------------------------------------------------- |
 | PyTorch               | 2.0.1                                              |
 | (PyTorch) Lightning\* | 2.0.x                                              |
-| **Lightning Habana**  | **1.0.1**                                          |
-| DeepSpeed             | Forked from v0.7.7 of the official DeepSpeed repo. |
+| **Lightning Habana**  | **1.1.0**                                          |
+| DeepSpeed             | Forked from v0.9.4 of the official DeepSpeed repo. |
```
Comment (Borda) [inline README.md:71]:
```suggestion
| DeepSpeed\*\*             | Forked from v0.9.4 of the official DeepSpeed repo. |
```
and lets add link where to download/install it

**Review by Borda — APPROVED** [review]

**Pull request merged**